backend/src/main/java/io/debezium/configserver/service/mongodb/MongoDbConnectorIntegrator.java

```diff
@@ -99,9 +96,9 @@ public class MongoDbConnectorIntegrator extends ConnectorIntegratorBase {
         MONGODB_PROPERTIES =  Collections.unmodifiableMap(additionalMetadata);
     }
 
-    protected ConnectionContext.MongoPrimary primary(MongoDbTaskContext context) throws Throwable {
-        ReplicaSet replicaSet = ReplicaSet.parse(context.getConnectionContext().hosts());
-        return context.getConnectionContext().primaryFor(replicaSet, context.filters(), (s, throwable) -> {
+    protected MongoDbConnection primary(MongoDbTaskContext context) throws Throwable {
```

Comment:
HOSTS and AUTO_DISCOVER_MEMBERS above are deprecated options. Should we remove them?  Also below in tests, should we better provide the `CONNECTION_STRING = Field.create("mongodb.connection.string")` instead?